src/esg_fastapi/api/versions/v1/routes.py

```diff
@@ -63,6 +67,35 @@ def query_instrumentor(query: ESGSearchQuery = Depends()) -> Generator[ESGSearch
 
 TrackedESGSearchQuery: ESGSearchQuery = Depends(query_instrumentor)
 
+_app_var: ContextVar[ClientApp] = ContextVar("_app_var")
+_search_client_var: ContextVar[SearchClient] = ContextVar("_search_client_var")
```

Comment:
Can you add some docs and tests for this? It's not immediately clear why we need ContextVars for these